### PR TITLE
Print failing bench jobs

### PR DESCRIPTION
**Pull request by SkySkimmer**
cf https://github.com/coq/coq/pull/16857

### CODE REVIEW

**Comment by SkySkimmer** [comment]
How do I get this to run?

**Comment by SkySkimmer** [comment]
Is there no CI here?

**Comment by Zimmi48** [comment]
Unfortunately, not yet sorry. The CI that we have is on the master branch. It both builds and deploys the Docker image to Heroku. So what we (Ali, Jason, and I) do for testing is that we have set up secret keys on our forks so that when pushing to the master branch on our forks, it will trigger a deployment, and this is how we can test PRs before merging them.

**Review by Zimmi48 — APPROVED** [review]
I will take care of test-deploying this one. (The code looks fine to me.)

**Comment by Zimmi48** [comment]
Actually, the code did not build. So what I'm currently deploying (https://github.com/Zimmi48/bot/actions/runs/3522703376/jobs/5905987230) is with a fix commit on top that I've also pushed here.

**Comment by Zimmi48** [comment]
> Unfortunately, not yet sorry. The CI that we have is on the master branch. It both builds and deploys the Docker image to Heroku. So what we (Ali, Jason, and I) do for testing is that we have set up secret keys on our forks so that when pushing to the master branch on our forks, it will trigger a deployment, and this is how we can test PRs before merging them.

In the future, we should set up CI on PRs to check the build, expect tests and formatting, and we could have a manual step to test-deploy.

File: src/actions.ml
Comment (SkySkimmer) [inline src/actions.ml:309]:
```suggestion
  let* failures = artifact_url "bench_failures" |> fetch_artifact in
```

**Comment by SkySkimmer** [comment]
Please redeploy

**Comment by Zimmi48** [comment]
In progress here: https://github.com/Zimmi48/bot/actions/runs/3523266243/jobs/5907222809

**Comment by SkySkimmer** [comment]
Tell me if you want another bench run

File: src/actions.ml
Comment (Alizter) [inline src/actions.ml:310]: Make this `let+` and you don't need `Lwt.return`.

**Comment by Zimmi48** [comment]
> Tell me if you want another bench run

I retriggered the webhook for the same bench instead.

**Comment by Zimmi48** [comment]
I've finally understood what was going on.

The text is generated twice (once for the comment, once for the bench Check summary). Here you only modified the latter. And indeed, the summary contains the info we were looking for:

https://github.com/coq/coq/pull/16857/checks?check_run_id=9642759097

**Comment by Zimmi48** [comment]
FWIW, here is the webhook you can redeliver to create a new comment without having to run a new bench: https://gitlab.com/coq/coq/-/hooks/352025/hook_logs/3826422401

**Comment by SkySkimmer** [comment]
Not super useful when I'm not the one deploying

**Comment by Zimmi48** [comment]
I'll redeploy and retrigger the webhook then. FWIW, setting up your fork to be able to deploy yourself would also be easy:

- Go to https://dashboard.heroku.com/account and reveal your API key.
- Go to https://github.com/SkySkimmer/bot/settings/secrets/actions and set `HEROKU_API_KEY` to this value and `HEROKU_EMAIL` to the email of your Heroku account.
- Make sure that Actions are enabled at https://github.com/Zimmi48/bot/settings/actions.
- Push to the `master` of your fork.

**Comment by Zimmi48** [comment]
Deploy in progress: https://github.com/Zimmi48/bot/actions/runs/3524519359/jobs/5910030778

**Comment by SkySkimmer** [comment]
It works

**Comment by Zimmi48** [comment]
Great. Feel free to directly merge this PR when ready (probably best once the Coq PR is merged).

**Comment by SkySkimmer** [comment]
Since we have
> Be resilient to missing timings/ artifacts 

order of merging shouldn't matter

**Comment by Zimmi48** [comment]
Right. Anyway, let's merge.

**Pull request merged**
